tests/components/myuplink/test_init.py

```diff
@@ -130,3 +164,57 @@ async def test_migrate_config_entry(
     assert mock_entry_v1_1.version == 1
     assert mock_entry_v1_1.minor_version == 2
     assert mock_entry_v1_1.unique_id == UNIQUE_ID
+
+
+async def test_oaut2_scope_failure(
+    hass: HomeAssistant,
+    mock_myuplink_client: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that an incorrect OAuth2 scope fails."""
+    with patch(
+        "homeassistant.components.myuplink.OAUTH2_SCOPES",
+        [
+            "wrong_scope",
+        ],
+    ):
+        await setup_integration(hass, mock_config_entry)
+
+    assert mock_config_entry.state is ConfigEntryState.SETUP_ERROR
```

Comment:
This is about the reauth right? Why don't we initialize with a different token in there?

Reply:
In the early days of this integration it was designed with a read-only scope. When the scope was extended to switch, select, and number entites we had to add the write scope and the natural way to do that was to trigger a reauth flow with a ConfigEntryAuthFailed exception, same exception as for other authentication failures. As I remember the test was designed to check the reauth flow by triggering with a scope error for no particular reason...

Reply:
Yes but why don't we add a config entry with that old token and have it happen naturally instead of patching the scopes?

Reply:
Changed now so we have three test of the reuth flow: Normal, old scope, account mismatch.

Reply:
Simplified test for bad scope as requested.